isobus/src/isobus_task_controller_client.cpp

```diff
@@ -114,16 +114,93 @@ namespace isobus
+	                                     std::uint32_t DDOPSize,
+	                                     std::uint8_t maxNumberBoomsSupported,
+	                                     std::uint8_t maxNumberSectionsSupported,
+	                                     std::uint8_t maxNumberChannelsSupportedForPositionBasedControl,
+	                                     bool reportToTCSupportsDocumentation,
+	                                     bool reportToTCSupportsTCGEOWithoutPositionBasedControl,
+	                                     bool reportToTCSupportsTCGEOWithPositionBasedControl,
+	                                     bool reportToTCSupportsPeerControlAssignment,
+	                                     bool reportToTCSupportsImplementSectionControl)
+	{
+		if (StateMachineState::Disconnected == get_state())
+		{
+			assert(nullptr != binaryDDOP); // Client will not work without a DDOP.
+			assert(0 != DDOPSize);
+			generatedBinaryDDOP.clear();
+			ddopStructureLabel.clear();
+			ddopLocalizationLabel.fill(0x00);
+			ddopUploadMode = DDOPUploadType::UserProvidedBinaryPointer;
+			userSuppliedBinaryDDOP = binaryDDOP;
+			userSuppliedBinaryDDOPSize_bytes = DDOPSize;
+			set_common_config_items(maxNumberBoomsSupported,
+			                        maxNumberSectionsSupported,
+			                        maxNumberChannelsSupportedForPositionBasedControl,
+			                        reportToTCSupportsDocumentation,
+			                        reportToTCSupportsTCGEOWithoutPositionBasedControl,
+			                        reportToTCSupportsTCGEOWithPositionBasedControl,
+			                        reportToTCSupportsPeerControlAssignment,
+			                        reportToTCSupportsImplementSectionControl);
+		}
+		else
+		{
+			// We don't want someone to erase our object pool or something while it is being used.
+			CANStackLogger::error("[TC]: Cannot reconfigure TC client while it is running!");
+		}
+	}
+
+	void TaskControllerClient::configure(const std::vector<std::uint8_t> &binaryDDOP,
+	                                     std::uint8_t maxNumberBoomsSupported,
+	                                     std::uint8_t maxNumberSectionsSupported,
+	                                     std::uint8_t maxNumberChannelsSupportedForPositionBasedControl,
+	                                     bool reportToTCSupportsDocumentation,
+	                                     bool reportToTCSupportsTCGEOWithoutPositionBasedControl,
+	                                     bool reportToTCSupportsTCGEOWithPositionBasedControl,
+	                                     bool reportToTCSupportsPeerControlAssignment,
+	                                     bool reportToTCSupportsImplementSectionControl)
+	{
+		if (StateMachineState::Disconnected == get_state())
+		{
+			assert(!binaryDDOP.empty()); // Client will not work without a DDOP.
+			ddopStructureLabel.clear();
+			ddopLocalizationLabel.fill(0x00);
+			generatedBinaryDDOP = binaryDDOP;
```

Comment:
Yeah I debated on that as well... ultimately it just means instead of re-using the `generatedBinaryDDOP ` variable, I'd need to store another pointer, but that seems fine I suppose. I can change it